rtl/eclrtl/rtldynfield.cpp

```diff
@@ -683,13 +709,41 @@ class CRtlFieldTypeDeserializer : public CInterfaceOf<IRtlFieldTypeDeserializer>
             }
         }
         info.fieldType &= ~RFTMserializerFlags;
-        return info.createRtlTypeInfo();
+        return info.createRtlTypeInfo(useBias);
+    }
+    void patchIndexFilePos()
+    {
+        if (callback && (base->fieldType & RFTMkind) == type_record)
+        {
+            // Yukky hack time
+            // Assumes that the fieldinfo is not shared...
+            // But that is also assumed by the code that cleans them up.
+            const RtlFieldInfo * const *fields = base->queryFields();
+            const RtlFieldInfo *lastField = nullptr;
+            unsigned idx = 0;
+            while (fields[idx])
+            {
+                lastField = fields[idx];
+                idx++;
+            }
+            assertex(lastField);
+            switch(lastField->type->fieldType & RFTMkind)
```

Comment:
minor: common up lastType = lastField->type

Reply:
Have you checked that the last field is ever non integer?
If that is true, then it would also be possible for the last field to be an integer, but not stored in the fileposition extra.

   index({unsigned id},{},'i')